libraries/WiFi/examples/WiFiClientConnect/WiFiClientConnect.ino

```diff
@@ -9,8 +9,22 @@
 */
 #include <WiFi.h>
 
-const char* ssid     = "your-ssid";
-const char* password = "your-password";
+// To use secrets please read the documentation at https://espressif-docs.readthedocs-hosted.com/projects/arduino-esp32/en/latest/guides/secrets.html
+#if __has_include("secrets.h")
```

Comment:
It is not an error when it does not exist - making it an error would instantly render all these examples unable to compile and not backward compatible. So not having secrets should be always ok. Having them is something on top of the default.
I will add some notification message instead of error.

Reply:
good point!